compiler/rustc_span/src/source_map.rs

```diff
@@ -463,6 +463,29 @@ impl SourceMap {
         self.span_to_string(sp, FileNameDisplayPreference::Remapped)
     }
 
+    /// Format the span location suitable for pretty printing anotations with relative line numbers
+    pub fn span_to_relative_line_string(&self, sp: Span, relative_to: Span) -> String {
+        if self.files.borrow().source_files.is_empty() || sp.is_dummy() || relative_to.is_dummy() {
+            return "no-location".to_string();
+        }
+
```

Comment:
I think we should `assert!(relative_to.contains(sp))`

Reply:
In the way it's currently used the assert would make sense, but I think usage where the two spans are separate but one is before the other would also be correct. The assert would panic though.

Reply:
yea that's true, but I think we'll also want to know if anyone starts doing that. I don't know how well that would be supported right now, so I'd rather not silently support it and get suprised in the future.